components/ide/jetbrains/gateway-plugin/src/main/kotlin/io/gitpod/jetbrains/gateway/GitpodConnectionProvider.kt

```diff
@@ -296,7 +296,11 @@ class GitpodConnectionProvider : GatewayConnectionProvider {
                                     )
                                     clientHandle.clientClosed.advise(connectionLifetime) {
                                         application.invokeLater {
-                                            connectionLifetime.terminate()
+                                            GlobalScope.launch {
+                                                // Delay for 5 seconds to wait the thinClient actually close
+                                                delay(5000)
```

Comment:
how do we know 5secs is enough, is there a triggere to which we can reliable bind instead?

Reply:
I don't think thinClient will block for 5 more seconds (unless UI freeze maybe?). From java Doc, there's no other public listeners available, I found `clientProcessTerminated` but it's private

<img width="1166" alt="image" src="https://github.com/user-attachments/assets/92f0a5e6-27c0-4cdf-9437-cb0fbfe38747">